engine/chain/internal/bhs/bhs_mock_test.go

```diff
@@ -0,0 +1,52 @@
+package bhs_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail/spv"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/jarcoal/httpmock"
+)
+
+const (
+	bhsURL   = "http://localhost:8080"
+	bhsToken = "mQZQ6WmxURxWz5ch"
+)
+
+func bhsMockVerify(response *chainmodels.MerkleRootsConfirmations, applyTimeout bool) *resty.Client {
+	transport := httpmock.NewMockTransport()
+	client := resty.New()
+	client.GetClient().Transport = transport
+
+	responder := func(req *http.Request) (*http.Response, error) {
+		if applyTimeout {
+			time.Sleep(100 * time.Millisecond)
+		}
+		if req.Header.Get("Authorization") != "Bearer "+bhsToken {
+			return httpmock.NewStringResponse(http.StatusUnauthorized, ""), nil
+		}
+		var reqBody []*spv.MerkleRootConfirmationRequestItem
+		_ = json.NewDecoder(req.Body).Decode(&reqBody)
```

Comment:
if you would accept a string then you won't need to make a json from some struct, and you won't need to handle (or skip handling) an error in such case.

Reply:
Right 👍